doc/modules/linear_model.rst

```diff
@@ -313,15 +314,75 @@ is correct, i.e. that the data are generated by this model.
 They also tend to break when the problem is badly conditioned
 (more features than samples).
 
-.. figure:: ../auto_examples/linear_model/images/sphx_glr_plot_lasso_model_selection_001.png
-    :target: ../auto_examples/linear_model/plot_lasso_model_selection.html
+.. figure:: ../auto_examples/linear_model/images/sphx_glr_plot_lasso_lars_ic_001.png
+    :target: ../auto_examples/linear_model/plot_lasso_lars_ic.html
     :align: center
     :scale: 50%
 
+Mathematical details
+""""""""""""""""""""
+
+The definition of AIC (and thus BIC) might differ in the literature. In this
+section, we give more information regarding the criterion computed in
+scikit-learn. The AIC criterion is defined as:
+
+.. math::
+    AIC = -2 \log(\hat{L}) + 2 d
+
+where :math:`\hat{L}` is the maximum likelihood of the model and
+:math:`d` is the number of parameters (as well referred to as degrees of
+freedom in the previous section).
+
+For a linear Gaussian model, the maximum log-likelihood is defined as:
+
+.. math::
+    \log(\hat{L}) = - \frac{n}{2} \log(2 \pi) - \frac{n}{2} \ln(\hat{\sigma}^2) - \frac{\sum_{i=1}^{n} (y_i - \hat{y_i})^2}{2\hat{\sigma}^2}
```

Comment:
```suggestion
    \log(\hat{L}) = - \frac{n}{2} \log(2 \pi) - \frac{n}{2} \ln(\hat{\sigma}^2) - \frac{\sum_{i=1}^{n} (y_i - \hat{y}_i)^2}{2\hat{\sigma}^2}
```
Should we write `\log(\hat{L}) = \min{\theta} - \frac{n}{2} \log(2 \pi) ..` to make it clear that the yhat is the one found by minimizing this likelihood, i.e. the unpenalized one? I think this is an important point. For instance, for linear models, all the yhats in this subsubsubsubsection are `X\hat{\beta}` with `\hat{\beta}` the OLS estimates.

In addition, I would use `\sigma` instead of `\hat{\sigma}`. It does't change the yhat, but is generally more correct.

Last point: This holds **in-sample** only. We could mention that above as advantage: We don't need a test set for AIC/BIC.

Reply:
> In addition, I would use \sigma instead of \hat{\sigma}. It does't change the yhat, but is generally more correct.

using `\sigma` does not make it obvious that this is an estimation, isn't it? The

> We don't need a test set for AIC/BIC.

I recall having mentioned it somewhere, maybe the new example?

Reply:
I improve the documentation for the last point regarding the criterion computation in-sample.

Reply:
> Using \sigma does not make it obvious that this is an estimation, isn't it?

Strictly speaking, MLE of a (linear) Gaussian model does not (need to) estimate sigma. We need sigma for AIC and don't even use the MLE estimate of it, but the Bessel corrected, unbiased estimated of sigma^2. Therefore, I would prefer to write `\sigma` instead of `\hat{\sigma}` in this case.